### ISSUE TITLE
Missing library on FreeBSD

### ISSUE THREAD

**Issue by pnorman** [include-what-you-use/include-what-you-use#225]
```
[pnorman@pippin /usr/home/pnorman/include-what-you-use/build]$ make
[  7%] Linking CXX executable include-what-you-use
/usr/bin/ld: cannot find -ldl
CC: error: linker command failed with exit code 1 (use -v to see invocation)
```

#134 seems related, and removing the `dl` requirement in CMakeLists allowed it to build, but an extra condition is probably required so it will work on both linux and freebsd

**Comment by kimgr** [comment]
It's been a while since I last did so, but I build IWYU on FreeBSD without much trouble.

CMakeLists.txt should only add `dl` to the link line if the system classifies as Linux... Any idea why your BSD would do so?

**Comment by pnorman** [comment]
> CMakeLists.txt should only add dl to the link line if the system classifies as Linux... Any idea why your BSD would do so?

It looks like for Clang 3.4, dl is being added for [all non-windows systems](https://github.com/include-what-you-use/include-what-you-use/blob/clang_3.4/CMakeLists.txt#L120-L130)

**Comment by kimgr** [comment]
OK, that makes sense. Since this is now fixed, I don't see that we'll do much more about it.

**Comment by kimgr** [comment]
I'm going to close this unless someone objects?

**Comment by kimgr** [comment]
This was fixed in IWYU 0.3, we're not aiming to backport it.
